File: forum/src/main/java/com/example/forum/config/JpaConfig.java
package com.example.forum.config;

import com.querydsl.core.util.StringUtils;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.domain.AuditorAware;
import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.core.userdetails.User;

import java.util.Optional;

@EnableJpaAuditing
@Configuration
public class JpaConfig {
    @Bean
    public AuditorAware<String> auditorAware() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        if (authentication == null || !authentication.isAuthenticated())
            return () -> Optional.of("INIT");

        User user = (User) authentication.getPrincipal();
        return () -> Optional.of(user.getUsername());
    }
}

File: migration-to-kotlin/src/main/kotlin/com/example/practice/config/JpaConfig.kt
package com.example.practice.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.domain.AuditorAware
import org.springframework.data.jpa.repository.config.EnableJpaAuditing
import java.util.Optional

@EnableJpaAuditing
@Configuration
class JpaConfig {
    @Bean
    fun auditorAware() : AuditorAware<String> {
        return AuditorAware { Optional.of("SEUNGSU") }
    }
}